### PR TITLE
[4.3] Smart Search: Hiding language taxonomy without multilanguage enabled

### PR DESCRIPTION
**Pull request by Hackwar**
Pull Request for Issue #39390.

### Summary of Changes
When the site is not multilanguage, the language taxonomy in Smart Search should not be displayed. Also, don't display the language '*'.

### Testing Instructions
Look at search results in the frontend and see the taxonomy "Language" on a monolingual site. Apply the patch and see the language disappear. Enable the languagefilter plugin and see the language taxonomy only apear for items which have a language defined.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: components/com_finder/tmpl/search/default_result.php [hunk 1]
```diff
@@ -105,8 +106,14 @@
...
-                <?php $branch = Taxonomy::getBranch($type); ?>
-                <?php if ($branch->state == 1 && in_array($branch->access, $user->getAuthorisedViewLevels())) : ?>
+                <?php
+                if ($type == 'Language' && (!Multilanguage::isEnabled() || (isset($taxonomy[0]) && $taxonomy[0]->title == '*'))) :
+                    continue;
+                endif;
+
+                $branch = Taxonomy::getBranch($type);
+
+                if ($branch->state == 1 && in_array($branch->access, $user->getAuthorisedViewLevels())) : ?>
```
Comment (Quy) [inline components/com_finder/tmpl/search/default_result.php]:
```suggestion
                <?php if ($type == 'Language' && (!Multilanguage::isEnabled() || (isset($taxonomy[0]) && $taxonomy[0]->title == '*'))) : ?>
                    <?php continue; ?>
                <?php endif; ?>
                <?php $branch = Taxonomy::getBranch($type); ?>
                <?php if ($branch->state == 1 && in_array($branch->access, $user->getAuthorisedViewLevels())) : ?>
```

Comment (Hackwar) [reply components/com_finder/tmpl/search/default_result.php]: I don't want to wrap each line in those PHP-tags, starting with the problem that that bloats the output with lots of empty lines. And I don't think it makes the code more readable either.

Comment (Quy) [reply components/com_finder/tmpl/search/default_result.php]: Per https://developer.joomla.org/coding-standards/php-code.html:

> Mixed language usage (e.g. at the layout files)
> 
> For layout files and all files where we use a mix of PHP and HTML (all PHP files in the `view/tmpl `and `layout` folder) we additionally wrap every line into a `<?php ... ?>` block and use the alternative syntax for control structures. This should make the code easier to read and make it easier to move blocks around without creating fatal errors due to missing `<?php ... ?>` tags.
>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 663f253a9f365189b1ae246d38241ae91a9d6fbd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39919">issues.joomla.org/tracker/joomla-cms/39919</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 663f253a9f365189b1ae246d38241ae91a9d6fbd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39919">issues.joomla.org/tracker/joomla-cms/39919</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39919">issues.joomla.org/tracker/joomla-cms/39919</a>.</sub>

**Comment by obuisard** [comment]
Thanks Hannes @Hackwar !

**Pull request merged**